src/main/scala/firrtl/passes/RemoveAccesses.scala

```diff
@@ -0,0 +1,166 @@
+package firrtl.passes
+
+import firrtl.ir._
+import firrtl.{WRef, WSubAccess, WSubIndex, WSubField}
+import firrtl.Mappers._
+import firrtl.Utils._
+import firrtl.WrappedExpression._
+import firrtl.Namespace
+import scala.collection.mutable
+
+
+/** Removes all [[firrtl.WSubAccess]] from circuit
+  */
+object RemoveAccesses extends Pass {
+  def name = "Remove Accesses"
+
+  /** Container for a base expression and its corresponding guard
+    */
+  case class Location(base: Expression, guard: Expression)
+
+  /** Walks a referencing expression and returns a list of valid references
+    * (base) and the corresponding guard which, if true, returns that base.
+    * E.g. if called on a[i] where a: UInt[2], we would return:
+    *   Seq(Location(a[0], UIntLiteral(0)), Location(a[1], UIntLiteral(1)))
+    */
+  def getLocations(e: Expression): Seq[Location] = e match {
+    case e: WRef => create_exps(e).map(Location(_,one))
+    case e: WSubIndex =>
+      val ls = getLocations(e.exp)
+      val start = get_point(e)
+      val end = start + get_size(tpe(e))
+      val stride = get_size(tpe(e.exp))
+      val lsx = mutable.ArrayBuffer[Location]()
+      var c = 0
+      for (i <- 0 until ls.size) {
+        if (((i % stride) >= start) & ((i % stride) < end)) {
+          lsx += ls(i)
+        }
+      }
+      lsx
+    case e: WSubField =>
+      val ls = getLocations(e.exp)
+      val start = get_point(e)
+      val end = start + get_size(tpe(e))
+      val stride = get_size(tpe(e.exp))
+      val lsx = mutable.ArrayBuffer[Location]()
+      var c = 0
+      for (i <- 0 until ls.size) {
+        if (((i % stride) >= start) & ((i % stride) < end)) { lsx += ls(i) }
+      }
+      lsx
+    case e: WSubAccess =>
+      val ls = getLocations(e.exp)
+      val stride = get_size(tpe(e))
+      val wrap = tpe(e.exp).asInstanceOf[VectorType].size
+      val lsx = mutable.ArrayBuffer[Location]()
+      var c = 0
+      for (i <- 0 until ls.size) {
```

Comment:
I think instead of using a var, you can just do

    val c = (i / stride) % wrap

at the start of each loop iteration.